src/app/components/upload-event/upload-event.component.ts

```diff
@@ -9,20 +11,75 @@ import { Event } from '../../models/event';
 })
 export class UploadEventComponent implements OnInit {
 
-  eventModel = new Event();
+  eventToUpload: Event;
+  eventForm: FormGroup;
+  noOfUploads = 10;
 
-  @ViewChild("eventForm") eventForm;
-  @ViewChild("userIdField") userIdInput : ElementRef;
 
-  constructor(private uploadEventService: UploadEventService) { }
+  @ViewChild('userIdInput') userIdInput: ElementRef;
+
+  constructor(private uploadEventService: UploadEventService, private fb: FormBuilder) {
+    this.createForm();
+  }
 
   ngOnInit() {
   }
 
+  createForm() {
+    this.eventForm = this.fb.group({
+      userId: ['', [Validators.required]],
+      userName: ['', [Validators.required]],
+      group: ['', [Validators.required]],
+      assetId: ['', [Validators.required, Validators.min(0)]],
+      assetTitle: ['', [Validators.required]],
+      startDate: ['', [Validators.required]],
+      endDate: ['', [Validators.required]]
+    }, {
+        validator: Validators.compose([
+          CustomValidators.dateLessThan('startDate', 'endDate', { 'loaddate': true })
+        ])
+      });
+  }
+
   onSubmit() {
-    this.uploadEventService.postEvent(this.eventModel).subscribe();
+    const startDate = new Date(this.eventForm.get('startDate').value);
+    const endDate = new Date(this.eventForm.get('endDate').value);
+    const differenceInDays = this.calculateDifferenceInDays(startDate, endDate) + 1;
+    console.log('Number of event uploads: ' + this.noOfUploads * differenceInDays);
+    for (let i = 0; i < differenceInDays; i++) {
+      for (let j = 0; j < this.noOfUploads; j++) {
+        const eventDate = this.calculateRandomEventTime(this.eventForm.get('startDate').value, i);
```

Comment:
These old school `for` loops can be replaced with something newer (and slightly hacky):
```
    [...Array(differenceInDays)].forEach((_, i) => {
      [...Array(this.noOfUploads)].forEach((_, j) => {
                
      });
    });
```